Various slight tweaks to README.md (#1)
dotlin-org/dotlin: README.md

Note the `@` before `const`. This is because `@const` is an annotation, not a keyword.
The Kotlin compiler does not support keywords in front of expressions at the parser level.

The difference is easy to remember: with any _declaration_ you must use `const`, and with any
_invocation_ you must use `@const`.

> Note that similar to in Dart, `@const` is not necessary when it's implied, e.g. by assigning to a `const val`.

Reviewer: ```suggestion
Note that similar as in Dart, `@const` is not necessary when it's implied, e.g. by assigning to a `const val`.
```
I think this is more correct?

Author: Changed `similar to in` to just `as in`, which I think flows a bit better anyway.

> ⚠️ Arguments of a constant creation must be constant expressions.

> Even though if you'd passed a reference of a _named_ top-level/static function

Reviewer: ```suggestion
Even though if you'd pass a reference to a _named_ top-level/static function
```
or
```suggestion
Even though if you would've passed a reference to a _named_ top-level/static function
```
Is correct I think :sweat_smile:

Author: Hmm, looking it up, I _think_ the way I currently have it ("if you had") is more "correct"? I think I've heard both ways before, though, so not sure.

Reviewer: Ahh yeah "if you had" is correct indeed, currently it says "you'd" with the "'d" :smile:

Then I guess the third option is the following, which is what I think you were going for?
```suggestion
Even though if you passed a reference of a _named_ top-level/static function
```
